src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -1798,6 +1798,40 @@
       <summary>Creates an ImplicitStackAllocArrayCreationExpressionSyntax node.</summary>
     </FactoryComment>
   </Node>
+  <Node Name="CollectionCreationExpressionSyntax" Base="ExpressionSyntax">
+    <Kind Name="CollectionCreationExpression"/>
+    <Field Name="OpenBracketToken" Type="SyntaxToken">
+      <Kind Name="OpenBracketToken"/>
+    </Field>
+    <Field Name="Elements" Type="SeparatedSyntaxList&lt;CollectionElementSyntax&gt;" AllowTrailingSeparator="true">
+      <PropertyComment>
+        <summary>SeparatedSyntaxList of CollectionElementSyntax representing the list of elements in the collection expression.</summary>
+      </PropertyComment>
+    </Field>
+    <Field Name="CloseBracketToken" Type="SyntaxToken">
+      <Kind Name="CloseBracketToken"/>
+    </Field>
+  </Node>
+  <AbstractNode Name="CollectionElementSyntax" Base="CSharpSyntaxNode" />
+  <Node Name="ExpressionElementSyntax" Base="CollectionElementSyntax">
+    <Kind Name="ExpressionElement"/>
+    <Field Name="Expression" Type="ExpressionSyntax" />
+  </Node>
+  <Node Name="SpreadElementSyntax" Base="CollectionElementSyntax">
+    <Kind Name="SpreadElement"/>
+    <Field Name="OperatorToken" Type="SyntaxToken">
+      <Kind Name="DotDotToken"/>
+    </Field>
+    <Field Name="Expression" Type="ExpressionSyntax" />
+  </Node>
+  <Node Name="DictionaryElementSyntax" Base="CollectionElementSyntax">
+    <Kind Name="DictionaryElement"/>
```

Comment:
Could this be a general expression of type KVP just like ranges?

Reply:
not sure waht this means.

Reply:
I mean something like `KeyValuePair<int, int> p = 1:1;`.



But even if it's only permitted for collection literals this makes the entire 'element' hierarchy redundant. Parser won't produce such trees anyways - slice pattern for example is just a subtype of pattern.

Reply:
> I mean something like

We could consider it.  But it makes many things very ambiguous.  For example `x ? y : z ...`  is `y` complete, or is it `y:z`?

Reply:
> But even if it's only permitted for collection literals this makes the entire 'element' hierarchy redundant.

That's a possibility, but i'm generally averse to it as it's just something that appears nowhere else.  

We could have this philosophy in a lot of things.  For example, we could say that all Argument-Calls just take expressions (instead of arguments), and that there are things like `named expressions` instead.  But we don't do that.  We say that if something is specific to a particular set of nodes, it should get a different type.

Reply:
Argument is an expression optionally precided by a name, literally. But a pair or spread are still expressions that happen to be permitted only in certain contexts even if it's specced using a separate grammar rule. 

I'd agree that this feels like overfitting the syntax into the existing forms but if anything that does simplify usage patterns using a more unified object model IMO.

Reply:
> But a pair or spread are still expressions

I don't see why.  Especially if they aren't allowed anywhere else in the language.  I don't see how that's different than "Argument is an expression optionally precided by a name".  They're all expressions with some slight additions, but only usable in a single context.

> but if anything that does simplify usage patterns using a more unified object model IMO.

I disagree.  For one thing, now people have to know that these expressions could not show up anywhere but here.  That's more difficult to reason about. For another, these are not actual expressions.  For example, they would not support GetTypeInfo, GetConversion, and the like.  From an API perspective, they would be odd items out.  

From teh lang perspective these are different.  We normally would make the syntax match the lang.  however, we do occasionally relax things at the syntax level (like allowing 'ref type'), but only if it comes with *bit* benefits (like error recovery).  That situation does not seem to apply here.

Reply:
Spread expression may be used as tuple splatting. M(.. tuple);

Pair expression maybe used as, well, a pair expression or something else, maybe. 

On reusing syntax I always bring up assignments in object initializer which is exactly what would happen with pairs here.

Reply:
Oh forget it, that's a range.

Reply:
Just to add some background from  [LDM-2022-03-09](https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-03-09.md)

>We might be able to say that .. on Index and int expressions creates a Range, and for any other type it performs a spread/splat operation, when in a spread/splat context. We didn't get too far into details around what constitutes such a context, but far enough that we think our future selves will be able to define such contexts.
>
>**Conclusion**
>We will explore option 3 when we get more in-depth on collection literals, and will keep .. as the syntax for slice patterns.